woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -2961,3 +2964,8 @@ def test_falsy_columns_in_partial_schema(falsy_names_df):
     assert new_df.ww.index == 0
     assert new_df.ww.time_index == ""
     assert new_df.ww.name == 0
+
+
+def test_nan_index_error(sample_df_pandas):
+    with pytest.raises(IndexError, match="Index contains null values"):
+        sample_df_pandas.ww.init(index="email")
```

Comment:
What do you think about including this in `test_check_index_errors`? I think that would be a more targeted test to confirm we get the right behavior from the `_check_index` function where this error actually gets raised, without needing to call the entire init process.

We actually should probably also have a test that confirms that `_check_index` function gets called on both `.init` and `.set_index` calls, if we don't already. That's a bit of a separate issue though.

Reply:
updated `test_check_index_errors` also added tests for `init` and `set_index`